src/Microsoft.AspNetCore.Server.HttpSys/AuthenticationHandler.cs

```diff
@@ -16,111 +16,94 @@ internal class AuthenticationHandler : IAuthenticationHandler
         private RequestContext _requestContext;
         private AuthenticationSchemes _authSchemes;
         private AuthenticationSchemes _customChallenges;
+        private AuthenticationScheme _scheme;
 
-        internal AuthenticationHandler(RequestContext requestContext)
-        {
-            _requestContext = requestContext;
-            _authSchemes = requestContext.Response.AuthenticationChallenges;
-            _customChallenges = AuthenticationSchemes.None;
-        }
-
-        public Task AuthenticateAsync(AuthenticateContext context)
+        public Task<AuthenticateResult> AuthenticateAsync()
         {
             var identity = _requestContext.User?.Identity;
-
-            foreach (var authType in ListEnabledAuthSchemes())
+            if (identity != null && identity.IsAuthenticated)
             {
-                var authScheme = authType.ToString();
-                if (string.Equals(authScheme, context.AuthenticationScheme, StringComparison.Ordinal))
+                foreach (var scheme in ListEnabledAuthSchemes())
                 {
-                    if (identity != null && identity.IsAuthenticated
-                        && string.Equals(authScheme, identity.AuthenticationType, StringComparison.Ordinal))
-                    {
-                        context.Authenticated(_requestContext.User, properties: null, description: null);
-                    }
-                    else
+                    if (string.Equals(scheme.ToString(), identity.AuthenticationType, StringComparison.Ordinal))
                     {
-                        context.NotAuthenticated();
+                        return Task.FromResult(AuthenticateResult.Success(new AuthenticationTicket(_requestContext.User, properties: null, authenticationScheme: _scheme.Name)));
```

Comment:
This says the scheme is "Negotiate", but you've registered "Windows". With a combine handler you can skip ListEnabledAuthSchemes and just always return the authenticated user.